### PR TITLE
Unify interface with crop-manager

### PR DESCRIPTION
**Pull request by talbaneth**

### CODE REVIEW

File: src/CharterManager.sol [hunk 1]
```diff
@@ -278,13 +278,15 @@ contract CharterManagerImp {
...
-    function flux(bytes32 ilk, address src, address dst, uint256 wad) external allowed(src) {
+    function flux(address gemJoin, address src, address dst, uint256 wad) external allowed(src) {
         address surp = getOrCreateProxy(src);
         address durp = getOrCreateProxy(dst);
 
-        VatLike(vat).flux(ilk, surp, durp, wad);
+        VatLike(vat).flux(ManagedGemJoinLike(gemJoin).ilk(), surp, durp, wad);
     }
 
+    function flee(address gemJoin) external {}
```
Comment (hexonaut) [inline src/CharterManager.sol:288]: Don't think we need an empty flee here. It's very specific to the crop join.

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: src/CharterManager.sol (hunk 1 above)
Comment (talbaneth) [reply src/CharterManager.sol:288]: You are right, but I think these will allow us to use a shared cdp manager layer if we ever want to.
We similaraly also have `onLiquidation` and `onVatFlux` that support the shared clipper.

File: src/CharterManager.sol [hunk 2]
```diff
@@ -278,13 +278,15 @@ contract CharterManagerImp {
         _validate(ilk, u, urp, dink, dart, rate, spot, _gate);
     }
 
-    function flux(bytes32 ilk, address src, address dst, uint256 wad) external allowed(src) {
+    function flux(address gemJoin, address src, address dst, uint256 wad) external allowed(src) {
         address surp = getOrCreateProxy(src);
```
Comment (hexonaut) [inline src/CharterManager.sol:282]: This needs `require(VatLike(vat).wards(crop) == 1, "ChaterManager/crop-not-authorized")`

Comment (talbaneth) [reply src/CharterManager.sol:282]: I don't think we need it as we don't do an external call to that address.
So it's similar to how we don't have it on frob now.

I do agree that we need to either not have it on both charter and crop join, or have it on both.

Comment (talbaneth) [reply src/CharterManager.sol:282]: Well actually we might not need to have it the same in both, as in crop join we do have the external call.
So I still think we can skip it here, as we could of just passed `ilk`, which wouldn't have any such sanity check.

Comment (hexonaut) [reply src/CharterManager.sol:282]: We do an external call to `ilk()`. Granted it's a view, so nothing can be modified.

**Review by hexonaut — APPROVED** [review]
After discussion this is all fine to me. Good to merge.

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Pull request merged**